### PR TITLE
HAWQ-120. Descriptive error message if the PXF plugin used has the ol…

### PR DESCRIPTION
**Pull request by hornn**
…d package name

### CODE REVIEW

File: pxf/pxf-service/src/test/java/org/apache/hawq/pxf/service/utilities/UtilitiesTest.java [hunk 1]
```diff
@@ -45,4 +41,22 @@ public void byteArrayToOctalString() throws Exception {
...
     }
+
+    @Test
+    public void createAnyInstanceOldPackageName() throws Exception {
+
+        InputData metaData = mock(InputData.class);
+
+        try {
+            Utilities.createAnyInstance(InputData.class,
+                    "com.pivotal.pxf.Lucy", metaData);
```
Comment (shivzone) [inline pxf/pxf-service/src/test/java/org/apache/hawq/pxf/service/utilities/UtilitiesTest.java]: haha good name

File: pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/utilities/Utilities.java [hunk 2]
```diff
@@ -30,8 +30,27 @@
...
+            cls = Class.forName(className);
+        } catch (ClassNotFoundException e) {
+            // in case the class name uses the old "com.pivotal.pxf" package
+            // name, recommend using the new package "org.apache.hawq.pxf".
+            if (className.startsWith("com.pivotal.pxf")) {
+                throw new Exception(
+                        "Class "
+                                + className
+                                + " doesn't not appear in classpath. "
+                                + "Try changing the plugin name to start with \"org.apache.hawq.pxf\"",
```
Comment (shivzone) [inline pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/utilities/Utilities.java]: If someone uses a custom plugin with their own implementation, it doesn't necessarily need to use this namespace right ? Can we instead mention "Plugins provided by PXF must start with org.apache.hawq.pxf"

**Comment by shivzone** [comment]
+1

File: pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/utilities/Utilities.java (hunk 2 above)
Comment (hornn) [inline pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/utilities/Utilities.java]: Done

File: pxf/pxf-service/src/test/java/org/apache/hawq/pxf/service/utilities/UtilitiesTest.java [hunk 3]
```diff
@@ -45,4 +48,26 @@ public void byteArrayToOctalString() throws Exception {
...
+
+        try {
+            Utilities.createAnyInstance(InputData.class,
+                    className, metaData);
+            fail("creating an instance should fail because the class doesn't exist in classpath");
+        } catch (Exception e) {
+            assertEquals(e.getClass(), Exception.class);
+            assertEquals(
+                    e.getMessage(),
+                    "Class " + className + " doesn't not appear in classpath. "
```
Comment (GodenYao) [inline pxf/pxf-service/src/test/java/org/apache/hawq/pxf/service/utilities/UtilitiesTest.java]: typo: "doesn't not"

Comment (hornn) [inline pxf/pxf-service/src/test/java/org/apache/hawq/pxf/service/utilities/UtilitiesTest.java]: Good catch! thanks!

**Pull request closed without merging**